### PR TITLE
update terminology before tearing into RFCs

### LINKED ISSUE #1990: Slot rotation vs Vote period
#### Problem

The RFCs conflate slot rotation and voting period.  It makes sense to schedule a single leader for multiple voting periods to amortize the cost of rotation, but to treat each voting period as a rollback point.

#### Proposed Solution

Fix the RFCs.

tag: @rob-solana @carllin

### PR DESCRIPTION
**Pull request by rob-solana**
Fixes #1990

### CODE REVIEW

File: book/src/terminology.md [hunk 1]
```diff
@@ -50,6 +50,10 @@ with a ledger interpretation that matches the leader's.
 
 A [ledger](#ledger) derived from common entries but then diverged.
 
+#### fortnight
```
Comment (aeyakovenko) [inline book/src/terminology.md]: don't use fortnight, `turn` is better.  but maybe to generic.  fornight is just my placeholder for a yet to be defined term

File: book/src/terminology.md [hunk 2]
```diff
@@ -178,6 +182,13 @@ A set of [transactions](#transaction) that may be executed in parallel.
 
 The role of a [fullnode](#fullnode) when it is validating the [leader's](#leader) latest [entries](#entry).
 
+#### vote
+
+A validator's affirmation that the PoH it has received has been verified.  A vote contains a hash of the updated state since the previous [voting period](#voting period).
```
Comment (garious) [inline book/src/terminology.md]: This was under `ledger vote`.

File: book/src/terminology.md [hunk 3]
```diff
@@ -178,6 +182,13 @@ A set of [transactions](#transaction) that may be executed in parallel.
...
+
+#### voting period
+
+The length of time in [ticks](#tick), between each [vote](#vote) submitted by a [validator](#validator).  Also called [slot](#slot).
```
Comment (garious) [inline book/src/terminology.md]: Do we only allow one vote per slot?  If slots are measured in ticks and a tick is about a second, how do we ensure subsecond finality?

File: book/src/terminology.md [hunk 4]
```diff
@@ -50,6 +50,10 @@ with a ledger interpretation that matches the leader's.
 
 A [ledger](#ledger) derived from common entries but then diverged.
 
+#### fortnight (placeholder name, could be "turn", "term?")
+
+The time (some number of [voting periods](#voting period)) during which a [leader](#leader) ingests transactions and produces [entries](#entry).
```
Comment (garious) [inline book/src/terminology.md]: This what I was calling a slot.

File: book/src/terminology.md (hunk 2 above)
Comment (rob-solana) [reply book/src/terminology.md]: ah

File: book/src/terminology.md (hunk 3 above)
Comment (rob-solana) [reply book/src/terminology.md]: if we separate voting and leader rotation period, we can vote as quickly as we like.  ticks are 100ms right now
https://github.com/solana-labs/solana/blob/777a0a858e59b3a3d8e7aa269d1a8941ce06c59b/src/poh_service.rs#L12

File: book/src/terminology.md (hunk 4 above)
Comment (rob-solana) [reply book/src/terminology.md]: yeah, and #1990 makes a new recommendation...

Comment (rob-solana) [reply book/src/terminology.md]: ok, do we want a one-word description of a "voting period"?  if so, how about "chunk", or "slice"? 

if we use "term" here, we can use "congress", or "ballot"

File: book/src/terminology.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ tracking its lifetime.
 
 #### block
 
-The [entries](#entry) generated within a [slot](#slot).
+A contiguous set of [entries](#entry), i.e. some segment of the PoH chain.
```
[the remark this answers is not in the document]
Comment (rob-solana) [reply book/src/terminology.md]: I'm using block here as a generic "slice of entries", nothing that has to do with votes, voting periods, or slots

[the remark this answers is not in the document]
Comment (garious) [reply book/src/terminology.md]: And I'm proposing that we get more specific with our definition of block such that we hit two birds with one stone. First, we adopt a definition that's similar to traditional blockchains (it's the thing all fullnodes agree is the next chunk of transactions). Second, that it fill this gap where we need a word for the set of entries that a fullnode is voting on.

[the remark this answers is not in the document]
Comment (rob-solana) [reply book/src/terminology.md]: how do you think it reads with "ballot" using "block" ?

[the remark this answers is not in the document]
Comment (garious) [reply book/src/terminology.md]: I'd write it this way: "ballot: A transaction sent by a validator to vote on a block."

[the remark this answers is not in the document]
Comment (garious) [reply book/src/terminology.md]: That's fine, but even as you just explained in there, you used vote as both a verb and ...an adjective?  I was just shooting for ballot == vote transaction.

[the remark this answers is not in the document]
Comment (rob-solana) [reply book/src/terminology.md]: yeah, I looked up ballot and was meh.  goal is single-word terms for block and slot, which I think we have now

**Pull request merged**